src/main/java/org/cbioportal/cdd/service/internal/ClinicalAttributeMetadataCache.java

```diff
@@ -124,21 +124,43 @@ private void validateAndResetCache() {
     */
     public void resetCache() {
         logger.info("resetCache(): refilling clinical attribute cache");
+
         Date dateOfCurrentCacheRefresh = new Date();
         List<ClinicalAttributeMetadata> latestClinicalAttributeMetadata = null;
         // latestOverrides is a map of study-id to list of overridden ClinicalAttributeMetadata objects
         Map<String, ArrayList<ClinicalAttributeMetadata>> latestOverrides = null;
+
+        // attempt to refresh ehcache stores seperately and store success status
+        boolean failedClinicalAttributeMetadataCacheRefresh = false;
+        boolean failedOverridesCacheRefresh = false;
         try {
-            latestClinicalAttributeMetadata = clinicalAttributesRepository.getClinicalAttributeMetadata();
-            latestOverrides = clinicalAttributesRepository.getClinicalAttributeMetadataOverrides();
+            clinicalAttributeMetadataPersistentCache.updateClinicalAttributeMetadataInPersistentCache();
         } catch (RuntimeException e) {
-            if (latestClinicalAttributeMetadata == null) {
-                logger.error("resetCache(): failed to pull clinical attributes from repository");
-            } else {
-                logger.error("resetCache(): failed to pull overrides from repository");
-            }
-            throw new FailedCacheRefreshException("Failed to refresh cache", e);
+            logger.error("resetCache(): failed to pull clinical attributes from repository. Error message returned: " + e.getMessage());
+            failedClinicalAttributeMetadataCacheRefresh = true;
+        }
+
+        try {
+            clinicalAttributeMetadataPersistentCache.updateClinicalAttributeMetadataOverridesInPersistentCache();
+        } catch (RuntimeException e) {
+            logger.error("resetCache(): failed to pull overrides from repository. Error message returned: " + e.getMessage());
+            failedOverridesCacheRefresh = true;
         }
+
+        // regardless of whether ehcache was updated with new data - use that data to populate modeled object caches
+        // ensures app starts up (between tomcat restarts) if TopBraid is down
+        logger.info("Loading modeled object cache from EHCache");
+        latestClinicalAttributeMetadata = clinicalAttributeMetadataPersistentCache.getClinicalAttributeMetadataFromPersistentCache();
```

Comment:
If cache fails to be updated, and then the persistent cache is corrupt, we get an exception here about failing to connect to TopBraid.